topcoffea/modules/datacard_tools.py

```diff
@@ -31,15 +31,22 @@ def to_hist(arr,name,zero_wgts=False):
     # NOTE:
     #   If we don't instantiate a new np.array here, then clipped will store a reference to the
     #   sub-array arr and when we modify clipped, it will propagate back to arr as well!
-    clipped = np.array(arr[1:-1])     # Strip off the under/overflow bins
-    clipped[-1] += arr[-1]  # Add the overflow bin to the right most bin content
-    nbins = len(clipped)
+    clipped=[]
+    for i in range(2): # first entry is sum(weight), second entry is sum(weight^2)
+        if arr[i] is not None:
+            clipped.append( np.array(arr[i][1:-1]))     # Strip off the under/overflow bins
+            clipped[i][-1] += arr[i][-1]  # Add the overflow bin to the right most bin content
+        else: 
+            clipped[i]=None
+
+
+    nbins = len(clipped[0])
     if zero_wgts:
         h = hist.Hist(hist.axis.Regular(nbins,0,nbins,name=name),storage=bh.storage.Weight())
-        h[...] = np.stack([clipped,np.zeros_like(clipped)],axis=-1) # Set the bin errors all to 0
+        h[...] = np.stack([clipped[0],np.zeros_like(clipped[0])],axis=-1) # Set the bin errors all to 0
     else:
-        h = hist.Hist(hist.axis.Regular(nbins,0,nbins,name=name))
-        h[...] = clipped
+        h = hist.Hist(hist.axis.Regular(nbins,0,nbins,name=name),storage=bh.storage.Weight())
+        h[...] = np.stack([clipped[0], clipped[1]],axis=-1)
```

Comment:
Since the instantiation of `h` is now identical in both branches of the `if` statement resolution, we should probably move it outside of the `if` block entirely.